Typewise-alert.hpp

```diff
@@ -0,0 +1,84 @@
+#ifndef TYPEWISE_ALERT_HPP
+#define TYPEWISE_ALERT_HPP
+
+class PASSIVE_COOLING
+{
+  public:
+  const int Lower_Limit = 0;
+  const int Upper_Limit = 35;
+};
+
+class HI_ACTIVE_COOLING
+{
+  public:
+  const int Lower_Limit = 0;
+  const int Upper_Limit = 45;
+};
+
+class MED_ACTIVE_COOLING
+{
+  public:
+  const int Lower_Limit = 0;
+  const int Upper_Limit = 40;
+};
+
+
+typedef enum {
+  NORMAL,
+  TOO_LOW,
+  TOO_HIGH
+} BreachType;
+
+
+typedef enum {
+  SENT,
+  NOT_SENT
+} AlertStatus;
+
+BreachType inferBreach(double value, double lowerLimit, double upperLimit);
+
+typedef enum {
+  TO_CONTROLLER,
+  TO_EMAIL,
+  TO_UNKNOWN
+} AlertTarget;
+
+
+template <class CoolingType>
```

Comment:
Instead of template class, inheritance could have been used.
Also the lines of code can be reduced.